test/run-drun/test-cost.mo

```diff
@@ -0,0 +1,52 @@
+import Prim "mo:⛔";
+
+import Cycles "cycles/cycles";
+
+actor client {
+  func print(t : Text) = Prim.debugPrint("client: " # t);
+
+  let ic00 = actor "aaaaa-aa" : actor {
+    create_canister : {
+      settings : ?{
+        controllers : ?[Principal];
+        compute_allocation : ?Nat;
+        memory_allocation : ?Nat;
+        freezing_threshold : ?Nat;
+      };
+    } -> async { canister_id : Principal };
+  };
+
+  public func go() : async () {
+    print(debug_show (Prim.costCreateCanister()) # " -- create canister cost");
+
+    if (Cycles.balance() == 0) {
+      await Cycles.provisional_top_up_actor(client, 3_000_000_000_000);
+      print("top up; balance = " # debug_show (Cycles.balance()));
+    } else {
+      print("already topped up; balance = " # debug_show (Cycles.balance()));
+    };
+
+    // let before = Cycles.balance();
+    printCycles();
+    let { canister_id } = await (with cycles = 500_000_000_000) ic00.create_canister({ settings = null });
```

Comment:
Output from the local replica:
```
38_461_538_461 -- create canister cost
already topped up; balance = 3_058_376_185_964
Cycles.balance()   = 3_058_376_185_964
Cycles.available() = 0
Cycles.refunded()  = 0
created canister id: be2us-64aaa-aaaaa-qaabq-cai
Cycles.balance()   = 2_558_375_756_574
Cycles.available() = 0
Cycles.refunded()  = 0
```
- The actual create canister cost is **38_461_538_461**, smaller than the [documented 500M](https://internetcomputer.org/docs/building-apps/essentials/gas-cost#canister-creation)
- `(with cycles = )` is required for the call to succeed
- extra cycles are not refunded, is this expected? Are they lost?

Reply:
I think both 2 and 3 are expected:

https://internetcomputer.org/docs/references/ic-interface-spec/#ic-management-canister-canister-creation

Reply:
Ad 2.
> [All cycles on this call are now the canister's initial cycles.](https://internetcomputer.org/docs/references/ic-interface-spec/#ic-management-canister-canister-creation)

Ok, so this might explain why cycles are not returned 👍 

Ad 1. @alexandru-uta found the answer: [The actual cost on a local replica is 1/13 of the documented value](https://internetcomputer.org/docs/building-apps/essentials/gas-cost#replication)

Reply:
@Kamirus @crusso -- shall we invest some time into running drun as app subnet? I can take a stab at it since my very initial test did not show any run-drun tests failing..